src/lib.rs

```diff
@@ -224,17 +246,28 @@ impl PublicKey {
         } else {
             bin
         };
-        if !ed25519::verify(bin, &self.key, &signature.signature) {
-            return Err(Error::InvalidSignature);
-        }
-        let trusted_comment_bin = signature.trusted_comment().as_bytes();
-        let mut global = Vec::with_capacity(signature.signature.len() + trusted_comment_bin.len());
-        global.extend_from_slice(&signature.signature[..]);
-        global.extend_from_slice(trusted_comment_bin);
-        if !ed25519::verify(&global, &self.key, &signature.global_signature) {
-            return Err(Error::InvalidSignature);
+        self.verify_ed25519(bin, signature)
+    }
+
+    /// Sets up a stream verifier that can be use iteratively.
+    pub fn verify_stream(&self) -> StreamVerifier {
+        let hasher = Blake2b::new(BLAKE2B_OUTBYTES);
+        StreamVerifier {
+            public_key: self,
+            hasher,
         }
-        Ok(())
+    }
+}
+
+impl<'a> StreamVerifier<'a> {
+    pub fn update(&mut self, buf: &[u8]) {
+        self.hasher.update(buf);
+    }
+
+    pub fn finalize(&mut self, signature: &Signature) -> Result<(), Error> {
+        let mut bin = vec![0u8; BLAKE2B_OUTBYTES];
+        self.hasher.finalize(&mut bin);
+        self.public_key.verify_ed25519(&bin, signature)
```

Comment:
cheers for the feedback :+1: I implemented it through `update` and `finalize`, to mimic the behaviour of a `Hasher`.
I can also set it up as a `Writer` - it's up to you !